src/Silverback.Integration/Messaging/Validation/MessageValidator.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Silverback.Messaging.Validation
+{
+    /// <summary>
+    ///     Validates the message and throws a <see cref="MessageValidationException" /> depending on the
+    ///     <see cref="MessageValidationMode" />.
+    /// </summary>
+    internal static class MessageValidator
+    {
+        /// <summary>
+        ///     Validates the message.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="validationMode">The message validation mode.</param>
+        /// <returns>
+        ///     A Tuple indicating whether the message is valid and the validation errors.
+        /// </returns>
+        /// <exception cref="MessageValidationException">
+        ///     The thrown exception when validation mode is
+        ///     <see cref="MessageValidationMode.ThrowException" />.
+        /// </exception>
+        public static (bool IsValid, string? ValidationErrors) CheckMessageIsValid(
```

Comment:
**(Just a commend, nothing to be changed.)**

Nice! I would have used an `out`  parameter out of habit, but using tuples is probably the way of the future.

Just a small consideration, I think the usage is still a bit more readable with an `out` parameter in these cases. But this might also be just because it's what we are most used to.

```csharp
(var success, var errors) Validator.CheckMessageIsValid(message);
if (success)
   Log(errors);

// ...vs..

if (!Validator.CheckMessageIsValid(message, out var errors))
   Log(errors);
```